pkg/api/types.go

```diff
@@ -205,23 +205,34 @@ type CustomNodesDNS struct {
 }
 
+// WindowsLicenseType represents Windows license type
+type WindowsLicenseType string
+
+const (
+	// WindowsLicenseTypeServer specifies that the image or disk that is being used was licensed server on-premises.
+	WindowsLicenseTypeServer WindowsLicenseType = "Windows_Server"
+	// WindowsLicenseTypeClient specifies that the image or disk that is being used was licensed client on-premises.
+	WindowsLicenseTypeClient WindowsLicenseType = "Windows_Client"
+)
+
 // WindowsProfile represents the windows parameters passed to the cluster
 type WindowsProfile struct {
-	AdminUsername             string            `json:"adminUsername"`
-	AdminPassword             string            `json:"adminPassword" conform:"redact"`
-	CSIProxyURL               string            `json:"csiProxyURL,omitempty"`
-	EnableCSIProxy            *bool             `json:"enableCSIProxy,omitempty"`
-	ImageRef                  *ImageReference   `json:"imageReference,omitempty"`
-	ImageVersion              string            `json:"imageVersion"`
-	WindowsImageSourceURL     string            `json:"windowsImageSourceURL"`
-	WindowsPublisher          string            `json:"windowsPublisher"`
-	WindowsOffer              string            `json:"windowsOffer"`
-	WindowsSku                string            `json:"windowsSku"`
-	WindowsDockerVersion      string            `json:"windowsDockerVersion"`
-	Secrets                   []KeyVaultSecrets `json:"secrets,omitempty"`
-	SSHEnabled                *bool             `json:"sshEnabled,omitempty"`
-	EnableAutomaticUpdates    *bool             `json:"enableAutomaticUpdates,omitempty"`
-	IsCredentialAutoGenerated *bool             `json:"isCredentialAutoGenerated,omitempty"`
+	AdminUsername             string              `json:"adminUsername"`
+	AdminPassword             string              `json:"adminPassword" conform:"redact"`
+	CSIProxyURL               string              `json:"csiProxyURL,omitempty"`
+	EnableCSIProxy            *bool               `json:"enableCSIProxy,omitempty"`
+	ImageRef                  *ImageReference     `json:"imageReference,omitempty"`
+	ImageVersion              string              `json:"imageVersion"`
+	WindowsImageSourceURL     string              `json:"windowsImageSourceURL"`
+	WindowsPublisher          string              `json:"windowsPublisher"`
+	WindowsOffer              string              `json:"windowsOffer"`
+	WindowsSku                string              `json:"windowsSku"`
+	WindowsDockerVersion      string              `json:"windowsDockerVersion"`
+	Secrets                   []KeyVaultSecrets   `json:"secrets,omitempty"`
+	SSHEnabled                *bool               `json:"sshEnabled,omitempty"`
+	EnableAutomaticUpdates    *bool               `json:"enableAutomaticUpdates,omitempty"`
+	IsCredentialAutoGenerated *bool               `json:"isCredentialAutoGenerated,omitempty"`
+	LicenseType               *WindowsLicenseType `json:"licenseType,omitempty"`
```

Comment:
Can this get added in alphabetically?
This really help reduce merge conflicts in code

Reply:
We just follow the current implementation. It seems like that we always add new property in the end. I think that the conflict is not caused by the sequence but by the different indent.

Reply:
> > @AbelHu Does AKS ever plan to use liscenseType of Windows_Client here? If not it might be good to add validation to fail if licenseType is set to Windows_Client.
> 
> AKS only uses `Windows_Server` and will validate this in AKS side. It seems like that SDK and API allows users to set it to `Windows_Client` so I think that it may be better to keep this option.

We don't support configuring Windows client nodes in aks-engine so I think it would be appropriate to add validation in aks-engine to dis-allow this even if it is possible in arm.

Reply:
> We just follow the current implementation. It seems like that we always add new property in the end. I think that the conflict is not caused by the sequence but by the different indent.

I know things aren't currently alphabetized everywhere but my ask is to start inserted new fields in alphabetically order.
It does help keep things neat and tidy.
I have been doing this whenever possible.

(also we have linting rules that make sure consistent indentation is caught in PRs)